components/support/error/tests/parse.rs

```diff
@@ -35,32 +32,36 @@ impl GetErrorHandling for Error {
     }
 }
 
-#[handle_error]
+#[handle_error(Error)]
 fn func() -> ::std::result::Result<String, ExternalError> {
     Err(Error{})
 }
 
-#[handle_error]
+#[handle_error(Error)]
 fn func2() -> Result<String, ExternalError> {
     Err(Error{})
 }
 
 type MyResult<T, E = ExternalError> = std::result::Result<T, E>;
 
-
-
-#[handle_error]
+#[handle_error(Error)]
 fn func3() -> MyResult<String> {
     Err(Error{})
 }
 
 type FullyAliasedResult<T = String, E = ExternalError> = std::result::Result<T, E>;
 
-#[handle_error]
+#[handle_error(Error)]
 fn func4() -> FullyAliasedResult {
     Err(Error{})
 }
 
+mod submodule {
+    #[super::handle_error(super::Error)]
+    fn func() -> ::std::result::Result<String, super::ExternalError> {
```

Comment:
nit: can we call this function in the `main` to test that the macro still returns what we expect it to, similar to to other functions?